### PR TITLE
Vector generate command

### PR DESCRIPTION
**Pull request by Westwooo**
The vector enrich-text command is used to produce vector indexable json docs form local data. The JSON will be of the form: 
```
{
     "id": "vectorx",
     "text":  "some chunk of text data",
     "vector": [0.187987123, ...., -0.0128379],
} 
```
It takes either a list of files output by ls:
```
ls | vector enrich-text
```

Or a string, this can be the contents of a single file, or user defined:
```
"embed this for me please" | vector enrich-text
open some-file.txt | vector enrich-text
```

The JSON output can be written directly to Couchbase: 
```
open ../some-file.txt | vector enrich-text | doc upsert
```

### CODE REVIEW

File: src/cli/vector_generate.rs [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+    }
+
+    let client =
+        Client::with_config(async_openai::config::OpenAIConfig::default().with_api_key(key));
+
+    let mut vector_count = 0;
+    let start = SystemTime::now();
+    for (i, batch) in batches.iter().enumerate() {
+        let batch_start = SystemTime::now();
+        print!("\rEmbedding batch {:?}/{} ", i + 1, batches.len());
```
Comment (chvck) [inline src/cli/vector_generate.rs:221]: Should this maybe be info! or debug!

Comment (Westwooo) [reply src/cli/vector_generate.rs:221]: Yeah should have been an info!, have made the change.

File: src/cli/vector_generate.rs [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+                            format!("failed to execute request: {}", e),
+                            "".to_string(),
+                            None,
+                            None,
+                            Vec::new(),
+                        ))
+                    }
+                },
+                () = ctrl_c_fut =>
+                     Err(ShellError::GenericError(
```
Comment (chvck) [inline src/cli/vector_generate.rs:242]: Is there a way to cancel the call via openai if ctrlc is pressed?

**Review by chvck — APPROVED** [review]
I think we probably want some iteration but this is a good point to build from.

File: src/cli/vector_generate.rs (hunk 2 above)
Comment (Westwooo) [reply src/cli/vector_generate.rs]: Not that is immediately obvious, I will make a note of this as an improvement but don't want to block this merge for that.

File: src/cli/util.rs [hunk 3]
```diff
@@ -354,6 +354,34 @@ pub fn duration_to_golang_string(duration: Duration) -> String {
     golang_string
 }
 
+pub fn read_openai_api_key(engine_state: &EngineState) -> Result<String, ShellError> {
```
[the remark this answers is not in the document]
Comment (Westwooo) [reply src/cli/util.rs:357]: Will do once this is merged

**Pull request merged**